non-CMake build does not install all headers

Hello!

I tried installing libSBML-5.19.0 with the standard&portable `./configure` `make` and `make install`, but obtained an installed package with some header files missing (notably, `sbml/math/*`).

The problem manifests even on simplests tests (also on basic examples from the docs), e.g. just:
```cpp
#include <sbml/SBMLTypes.h>
int main ()
{
  return 0;
}
```
does not build; the compiler reports:
```
In file included from /[...]/libSBML-install/include/sbml/SBMLDocument.h:307,
                 from /[...]/libSBML-install/include/sbml/SBMLTypes.h:4 ,
                 from includetest.cpp:1:
/[...]/libSBML-install/include/sbml/SBMLTransforms.h:56:10: fatal error: sbml/math/ASTNode.h: No such file or directory
   56 | #include <sbml/math/ASTNode.h>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
The whole `include/sbml/math` subdirectory is indeed missing in the installed files. I guess this is caused by some math structure changes around 2 years ago, relevant blame may include this: https://github.com/sbmlteam/libsbml/blame/development/src/sbml/math/Makefile.in#L146 . Recent git versions produce the same behavior.

I didn't find a simple solution; just adding back the `install-headers` target to `src/sbml/math/Makefile.in` fixes the installation of the headers, but the resulting library does not contain the required symbols:
```
/usr/bin/ld: /[...]/libSBML-install/lib//libsbml.so: undefined reference to `FormulaTokenizer_free'
/usr/bin/ld: /[...]/libSBML-install/lib//libsbml.so: undefined reference to `ASTNode::setParentSBMLObject(SBase*)'
/usr/bin/ld: /[...]/libSBML-install/lib//libsbml.so: undefined reference to `ASTNode::getChild(unsigned int) const'
[...around 300 others...]
```

Is there any easy fix other than using CMake (where the build scripts have been properly updated)? That said, I don't have any particular problem with using CMake in this case and will happily use it, but it would be great if the build docs could be updated to reflect the fact that the "traditional" installation scriptage is incomplete at the moment.

Thanks!

(cc @laurentheirendt )

--- a/src/sbml/math/Makefile.in
+++ b/src/sbml/math/Makefile.in
@@ -116,15 +116,14 @@ distfiles = $(sources) $(headers) Makefile.in
 # The default action is to remake everything.  Those rules which are not
 # defined below are defined in makefile-common-actions.mk.
 
-all: Makefile 
-# default
+all: Makefile default
 
 
 # -----------------------------------------------------------------------------
 # Checking.
 # -----------------------------------------------------------------------------
 
-#check: check-recursive
+check: check-recursive
 
 
 # -----------------------------------------------------------------------------
@@ -143,39 +142,24 @@ tags: etags ctags
 # in 'all' as a default to make sure files get compiled if someone does
 # 'make install' from the top level without first doing a 'make'.
 
-install:
+install: all install-headers
 
-uninstall:
+installcheck: all installcheck-headers
 
-installcheck:
-
-#install: all install-headers
-
-#installcheck: all installcheck-headers
-
-#uninstall: uninstall-headers
+uninstall: uninstall-headers
 
 
 # -----------------------------------------------------------------------------
 # Cleaning.
 # -----------------------------------------------------------------------------
 
-clean: clean-normal
+clean: clean-normal clean-recursive
 
-distclean: distclean-normal
+distclean: distclean-normal distclean-recursive
 
-mostlyclean: mostlyclean-normal
+mostlyclean: mostlyclean-normal mostlyclean-recursive
 
-maintainer-clean: maintainer-clean-normal
-
-
-#clean: clean-normal clean-recursive
-
-#distclean: distclean-normal distclean-recursive
-
-#mostlyclean: mostlyclean-normal mostlyclean-recursive
-
-#maintainer-clean: maintainer-clean-normal maintainer-clean-recursive
+maintainer-clean: maintainer-clean-normal maintainer-clean-recursive
 
 
 # -----------------------------------------------------------------------------

